lib/inspec/profile.rb

```diff
@@ -422,7 +421,24 @@ def check # rubocop:disable Metrics/AbcSize, Metrics/CyclomaticComplexity, Metri
         warn.call(sfile, sline, nil, id, "Control #{id} has impact > 1.0") if control[:impact].to_f > 1.0
         warn.call(sfile, sline, nil, id, "Control #{id} has impact < 0.0") if control[:impact].to_f < 0.0
         warn.call(sfile, sline, nil, id, "Control #{id} has no tests defined") if control[:checks].nil? or control[:checks].empty?
-      }
+
+        # Check for duplicate IDs within the same profile
+        rule = control[:rule_obj]
+        next unless Inspec::Rule.merge_count(rule) > 0
+        profile_id = Inspec::Rule.profile_id(rule)
+        merges = Inspec::Rule.merge_changes(rule)
+        merges = merges.select do |merge_location|
```

Comment:
It looks like y'all use the weirich method for blocks (`do`/`end` for side-effects, `{}` for result—something I strongly advocate for), so this select should use `{}`.

Reply:
I don't think we have any set pattern project-wide. I typically use do/end for multiline blocks, and `{}` for one-liners.